IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ElementsHandler.cs

```diff
@@ -91,5 +92,21 @@ public bool ContainsText(string text)
                 return false;
             }
         }
+
+        public IWebElement GetSingleExistingElement(Func<IWebElement, bool> predicate)
+        {
+            IWait<IWebDriver> wait = Wait;
+            wait.IgnoreExceptionTypes(typeof(NoSuchElementException));
+            return wait.Until(d =>
+            {
+                var foundElems = d.FindElements(Locator).Where(predicate).ToList();
+                if(foundElems.Count != 1)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(predicate), 
+                        "The search criteria did not return exactly one result.");
+                }
+                return foundElems.Single();
```

Comment:
nit: it is slightly better perf to use the indexer rather than the LINQ method here since you have already checked the count.
```suggestion
                return foundElems[0];
```

Reply:
@Keboo Updated with latest commit. Also had to make some logic changes to capture an alternative use case where something could fail. Wasn't sure if you wanted to review the udpates.